docs(behavior_path_planner): update goal planner docs (#3690)
autowarefoundation/autoware.universe: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md

PullOverPlannerBase --o GoalPlannerModule
GoalSeacherBase --o GoalPlannerModule
FixedGoalPlannerBase --o GoalPlannerModule

PullOverPath --o PullOverPlannerBase
GoalCandidates --o GoalSeacherBase

@enduml
```

## start condition

> Each is activated when all conditions are met.

Reviewer: ```suggestion
Either one is activated when all conditions are met.
```
??

Author: tnaks!!

### fixed_goal_planner

- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.

> - route is set with `allow_goal_modification=false` which is the default value.

Reviewer: ```suggestion
- Route is set with `allow_goal_modification=false` by default.
```

<img src="https://user-images.githubusercontent.com/39142679/237929955-c0adf01b-9e3c-45e3-848d-98cf11e52b65.png" width="600">

### rough_goal_planner

#### pull over on road lane

- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.

> - Route is set with `allow_goal_modification=true`

Reviewer: ```suggestion
- Route is set with `allow_goal_modification=true` .
```

> - route is set with `allow_goal_modification=true`

Reviewer: ```suggestion
- Route is set with `allow_goal_modification=true`
```

>   - we can set this option with [SetRoute](https://github.com/autowarefoundation/autoware_adapi_msgs/blob/main/autoware_adapi_v1_msgs/routing/srv/SetRoute.srv#L2) api service.

Reviewer: ```suggestion
  - We can set this option with [SetRoute](https://github.com/autowarefoundation/autoware_adapi_msgs/blob/main/autoware_adapi_v1_msgs/routing/srv/SetRoute.srv#L2) api service.
```

>   - we support `2D Rough Goal Pose` with the key bind `r` in rviz, but in the future there will be a panel of tools to manipulate various Route API from rviz.

Reviewer: ```suggestion
  - We support `2D Rough Goal Pose` with the key bind `r` in RViz, but in the future there will be a panel of tools to manipulate various Route API from RViz.
```

- ego-vehicle is in the same lane as the goal.

<img src="https://user-images.githubusercontent.com/39142679/237929950-989ca6c3-d48c-4bb5-81e5-e8d6a38911aa.png" width="600">

#### pull over on shoulder lane

- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.

> - Goal is set in the `road_shoulder`

Reviewer: ```suggestion
- Goal is set in the `road_shoulder`.
```

<img src="https://user-images.githubusercontent.com/39142679/237929941-2ce26ea5-c84d-4d17-8cdc-103f5246db90.png" width="600">

> ### finish condition

Reviewer: ```suggestion
## finish condition
```
If aligned with "## start condition"

> - The distance to the goal from your vehicle is lower than threshold (default: < `1m`)

Reviewer: ```suggestion
- The distance to the goal from your vehicle is lower than threshold (default: < `1m`).
```
